src/core/json/json_like.rs

```diff
@@ -1,35 +1,46 @@
 use std::collections::HashMap;
 
-pub trait JsonLikeOwned: for<'json> JsonLike<'json> {}
-impl<T> JsonLikeOwned for T where T: for<'json> JsonLike<'json> {}
+/*pub trait JsonLikeOwned: for<'json> JsonLike<'json> {}
+impl<T> JsonLikeOwned for T where T: for<'json> JsonLike<'json> {}*/
 
 /// A trait for objects that can be used as JSON values
-pub trait JsonLike<'a>: Sized {
-    type JsonObject: JsonObjectLike<'a, Value = Self>;
+pub trait JsonLike: Sized {
+    type JsonObject<'a>: JsonObjectLike<'a>
+    where
+        Self: 'a;
+    type Output<'a>: JsonLike
+    where
+        Self: 'a;
```

Comment:
@meskill `Output` needs to be of type `Self` here not any JsonLike. That constraint is the responsibility of JsonLike and not of the implementor. The implementor needs to ensure it matches the trait.